src/obu.rs

```diff
@@ -2053,22 +2065,12 @@ unsafe fn parse_obus(
             }
             // ensure that the reference is writable
             debug_assert!(rav1d_ref_is_writable(c.frame_hdr_ref) != 0);
-            let frame_hdrs =
-                (*c.frame_hdr_ref).data as *mut DRav1d<Rav1dFrameHeader, Dav1dFrameHeader>;
-            memset(
-                frame_hdrs as *mut c_void,
-                0,
-                ::core::mem::size_of::<DRav1d<Rav1dFrameHeader, Dav1dFrameHeader>>(),
-            );
-            c.frame_hdr = &mut (*frame_hdrs).rav1d;
-            (*c.frame_hdr).temporal_id = temporal_id;
-            (*c.frame_hdr).spatial_id = spatial_id;
-            let res = parse_frame_hdr(c, &mut gb);
-            if res.is_err() {
-                writeln!(c.logger, "Error parsing frame header");
-                c.frame_hdr = 0 as *mut Rav1dFrameHeader;
-                return Err(EINVAL);
-            }
+            let drav1d_frame_hdr_ptr = (*c.frame_hdr_ref)
+                .data
+                .cast::<DRav1d<Rav1dFrameHeader, Dav1dFrameHeader>>();
+            drav1d_frame_hdr_ptr.write(drav1d_frame_hdr);
```

Comment:
This implies that `drav1d_frame_hdr_ptr` is aligned correctly for the type, which we don't (yet) enforce. Totall ok with this for now, just a thing to keep in mind in terms of unsafety. This write call is unsafe so we will address this before removing it later after updating the allocation itself.

Reply:
Well we've been dereferencing and holding references to it before, which is UB if it's not properly aligned.  I used `.write` here vs. a normal write to a dereference just so the previous uninitialized value isn't dropped since it's uninitialized.  But it'll all go away once we make it an `Arc` so it should be fine eventually.

Reply:
Yep, just an aside.